### PR TITLE
Adapt event creator to new event model

### PR DESCRIPTION
**Pull request by Ef55**
I will leave updates on how the event creator looks.
Feel free to mention any UI improvement you imagine.

### CODE REVIEW

**Comment by Ef55** [comment]
Current look:
Screenshot_1616151128

File: app/src/androidTest/java/com/github/onedirection/EventCreatorTest.java [hunk 1]
```diff
@@ -54,45 +68,95 @@ public void setUp() {
...
+    public void eventDateCanSpecified() {
+        final LocalDate date = LocalDate.of(1000, 10, 1);
 
+        Intent intent = new Intent(ApplicationProvider.getApplicationContext(), EventCreator.class);
+        EventCreator.putDateExtra(intent, date);
 
-        //onView(allOf(instanceOf(TextView.class), withParent(withId(R.id.toolbar))))
-        //       .check(matches(withText(R.string.menu_calendar)));
-        //onView(withIndex(withId(R.id.calendarView), 0)).check(matches(isDisplayed()));
+        try (ActivityScenario<EventCreator> scenario = ActivityScenario.launch(intent)) {
```
Comment (hugolan) [inline app/src/androidTest/java/com/github/onedirection/EventCreatorTest.java:157]: nice use of try didn't thought about that

File: app/src/main/java/com/github/onedirection/EventCreator.java [hunk 2]
```diff
@@ -1,51 +1,244 @@
...
+                // TODO: update event in db
+            } else {
+                // TODO: put event in db
             }
+
+            viewEvent(event);
         });
     }
 
-    public void validateEvent(View view) {
```
Comment (hugolan) [inline app/src/main/java/com/github/onedirection/EventCreator.java:31]: kHPxj2F

do you really needed to refactor all my code? I mean it is nicer like this but let me do it next time

**Review by hugolan — APPROVED** [review]

File: app/src/main/java/com/github/onedirection/EventCreator.java (hunk 2 above)
Comment (Ef55) [reply app/src/main/java/com/github/onedirection/EventCreator.java:31]: Well, I had to refactor some parts anyway (as it was the goal of this PR), and I ended up doing some other refactoring I wouldn't normally do on someone's else code in the process, so sorry about that.

**Comment by Ef55** [comment]
This PR is now «dependent» on PR #66, and should as such be reviewed/merged after that one.

**Review by justagoodguy-omega — APPROVED** [review]
Truly delicious code to read, ligma

**Comment by Ef55** [comment]
<details> 
<summary>Current look: </summary>

</details>

**Pull request merged**